### PR TITLE
Add generic user provider and postgresql user provider

### PR DESCRIPTION
**Pull request by lpatte**
Generic user provider is for engine : 

- Cassandra
- Kafka
- KafkaConnect
- Mysql

### CODE REVIEW

File: ovh/resource_cloud_project_database_user.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+				Type:        schema.TypeString,
+				Description: "Name of the user",
+				ForceNew:    true,
+				Required:    true,
+			},
+
+			//Computed
+			"created_at": {
+				Type:        schema.TypeString,
+				Description: "Date of the creation of the cluster",
```
Comment (ybriffa) [inline ovh/resource_cloud_project_database_user.go:52]: wrong description

File: ovh/resource_cloud_project_database_postgresql_user.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+				Type:        schema.TypeList,
+				Description: "Roles the user belongs to",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			//Computed
+			"created_at": {
+				Type:        schema.TypeString,
+				Description: "Date of the creation of the cluster",
```
Comment (ybriffa) [inline ovh/resource_cloud_project_database_postgresql_user.go:53]: wrong description

File: website/docs/d/cloud_project_database_users.html.markdown [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  cluster_id   = "ZZZ"
+}
+```
+
+## Argument Reference
+
+* `service_name` - The id of the public cloud project. If omitted,
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+* `engine` - The database engine you want to deploy. To get a full list of available engine visit.
```
Comment (ybriffa) [inline website/docs/d/cloud_project_database_users.html.markdown:28]: wrong description

File: website/docs/r/cloud_project_database_postrgesql_user.html.markdown [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `service_name` - The id of the public cloud project. If omitted,
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+* `engine` - The database engine you want to deploy. To get a full list of available engine visit.
```
Comment (ybriffa) [inline website/docs/r/cloud_project_database_postrgesql_user.html.markdown:38]: wrong description

File: website/docs/r/cloud_project_database_user.html.markdown [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `service_name` - The id of the public cloud project. If omitted,
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+* `engine` - The database engine you want to deploy. To get a full list of available engine visit.
```
Comment (ybriffa) [inline website/docs/r/cloud_project_database_user.html.markdown:37]: wrong description

**Review by ybriffa — CHANGES_REQUESTED** [review]

**Review by ybriffa — APPROVED** [review]

File: website/docs/d/cloud_project_database_postgresql_user.html.markdown [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+# cloud_project_database_postgresql_user (Data Source)
+
+Use this data source to get information about a user of a postgresql cluster associated with a public cloud project.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_databases_user" "user" {
```
Comment (scraly) [inline website/docs/d/cloud_project_database_postgresql_user.html.markdown:16]:
```suggestion
data "ovh_cloud_project_database_postgresql_user" "pguser" {
```

File: website/docs/d/cloud_project_database_postgresql_user.html.markdown [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  service_name  = "XXX"
+  cluster_id    = "YYY"
+  user_id =     = "ZZZ"
+}
```
Comment (scraly) [inline website/docs/d/cloud_project_database_postgresql_user.html.markdown:20]:
```suggestion
}

output "pguser_roles" {
  value = data.ovh_cloud_project_database_postgresql_user.pguser.roles
}
```

File: website/docs/d/cloud_project_database_user.html.markdown [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_databases_user" "user" {
+  service_name  = "XXX"
+  engine	      = "YYY"
+  cluster_id    = "ZZZ"
+  user_id =     = "UUU"
+}
```
Comment (scraly) [inline website/docs/d/cloud_project_database_user.html.markdown:21]:
```suggestion
}

output "user_name" {
  value = data.ovh_cloud_project_database_user.user.name
}
```

File: website/docs/d/cloud_project_database_user.html.markdown [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+# cloud_project_database_user (Data Source)
+
+Use this data source to get information about a user of a database cluster associated with a public cloud project.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_databases_user" "user" {
```
Comment (scraly) [inline website/docs/d/cloud_project_database_user.html.markdown:16]:
```suggestion
data "ovh_cloud_project_database_user" "user" {
```

File: website/docs/d/cloud_project_database_user.html.markdown [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## Argument Reference
+
+* `service_name` - The id of the public cloud project. If omitted,
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+* `engine` - The engine of the database cluster you want user information. To get a full list of available engine visit :
+[public documentation](https://docs.ovh.com/gb/en/publiccloud/databases).\
+Aivalable engines for this resource (other have specific resource):
```
Comment (scraly) [inline website/docs/d/cloud_project_database_user.html.markdown:31]:
```suggestion
Available engines for this resource (other have specific resource):
```

File: website/docs/d/cloud_project_database_users.html.markdown [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+Use this data source to get the list of users of a database cluster associated with a public cloud project.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_databases_users" "users" {
+  service_name = "XXXX"
+  engine	   = "YYYY"
+  cluster_id   = "ZZZ"
+}
```
Comment (scraly) [inline website/docs/d/cloud_project_database_users.html.markdown:20]:
```suggestion
}

output "user_ids" {
  value = data.ovh_cloud_project_database_users.users.user_ids
}
```

File: website/docs/d/cloud_project_database_users.html.markdown [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+# cloud_project_database_users (Data Source)
+
+Use this data source to get the list of users of a database cluster associated with a public cloud project.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_databases_users" "users" {
```
Comment (scraly) [inline website/docs/d/cloud_project_database_users.html.markdown:16]:
```suggestion
data "ovh_cloud_project_database_users" "users" {
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: website/docs/d/cloud_project_database_postgresql_user.html.markdown [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+* `cluster_id` - Cluster ID
+
+* `name` - Name of the user.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `created_at` - Date of the creation of the user.
+* `id` - Public Cloud Database Service ID.
```
Comment (scraly) [inline website/docs/d/cloud_project_database_postgresql_user.html.markdown:41]:
```suggestion
* `id` - ID of the user.
```

File: website/docs/d/cloud_project_database_user.html.markdown [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+* `cluster_id` - Cluster ID
+
+* `name` - Name of the user.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `created_at` - Date of the creation of the user.
+* `id` - Public Cloud Database Service ID.
```
Comment (scraly) [inline website/docs/d/cloud_project_database_user.html.markdown:50]:
```suggestion
* `id` - ID of the user.
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — APPROVED** [review]

**Comment by scraly** [comment]
Thanks :)

**Pull request merged**